enhancements/ingress/operator-resource-configuration.md

```diff
@@ -0,0 +1,676 @@
+   - **A**: Out of scope for initial implementation, but API should not preclude it
+
+2. **Q**: Should we add validation for minimum resource values?
+   - **A**: Start with warnings/documentation, consider hard validation if issues arise
+
+3. **Q**: Should this apply to all IngressControllers or only the default?
+   - **A**: Initial implementation only default, but API supports any IngressController
+
+4. **Q**: How do we handle the operator modifying its own deployment safely?
+   - **A**: Use owner references carefully, reconcile loop with backoff
+
+### Test Plan
+
+#### Unit Tests
+
+- **API conversion tests**: v1 ↔ v1alpha1 conversion correctness
+- **Controller reconciliation logic**: Mock deployment updates
+- **Resource requirement validation**: Edge cases and invalid inputs
+- **Default value handling**: Ensure defaults applied correctly
+
+Coverage target: >80% for new code
+
+#### Integration Tests
+
+- **API server integration**: v1alpha1 CRD registration and serving
+- **Conversion webhook**: Automatic conversion between versions
+- **Controller watches**: IngressController changes trigger reconciliation
+
+#### E2E Tests
+
+- **Create IngressController with operatorResourceRequirements**
+  - Verify operator deployment is updated with correct resources
+  - Verify operator continues functioning normally
+  
+- **Update existing IngressController to add resource requirements**
+  - Verify rolling update occurs
+  - Verify no disruption to router functionality
+  
+- **Remove resource requirements (revert to defaults)**
+  - Verify deployment reverts to default values
+  
+- **Upgrade scenario tests**
+  - Upgrade from version without feature to version with feature
+  - Verify existing IngressControllers continue working
+  - Verify v1 API remains functional
+  
+- **Downgrade scenario tests**
+  - Downgrade from version with v1alpha1 to version without
+  - Verify graceful degradation (v1alpha1 fields ignored)
+
+#### Manual Testing
+
+- Test in resource-constrained environments (e.g., single-node)
+- Verify QoS class changes as expected (None → Burstable → Guaranteed)
+- Test with various resource configurations (very low, very high)
+- Test operator behavior when limits are hit (OOMKill, CPU throttling)
+- Test with multiple IngressController instances
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview (v1alpha1)
+
+- [x] Feature implemented behind feature gate
+- [x] Unit and integration tests passing
+- [x] E2E tests passing in CI
+- [x] Documentation published in OpenShift docs
+- [x] Enhancement proposal approved
+- [ ] Feedback collected from at least 3 early adopters
+- [ ] Known issues documented
+
+#### Tech Preview -> GA (promotion to v1)
+
+This section describes criteria for graduating from v1alpha1 to v1 (stable API).
+
+- [ ] Sufficient field testing (2+ minor releases in Tech Preview)
+- [ ] No major bugs reported for 2 consecutive releases
```

Comment:
This is an unusual requirement for OpenShift.  For a feature like this, we would usually introduce as Tech Preview and graduate to GA in the same release development cycle.